### PR TITLE
Loosen up equivalence checks for linked file quickinfo

### PR DESCRIPTION
**Pull request by rchande**
Fixes TFS #1158768

Can you take a look @Pilchie @jasonmalinowski @dpoeschl @balajikris @basoundr?

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
:+1:

File: src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs [hunk 1]
```diff
@@ -104,10 +104,10 @@ internal abstract partial class AbstractSemanticQuickInfoProvider : AbstractQuic
 
             // We calculate the set of supported projects
             candidateResults.Remove(bestBinding);
-
             foreach (var candidate in candidateResults)
             {
-                if (!candidate.Item3.SequenceEqual(bestBinding.Item3, LinkedFilesSymbolEquivalenceComparer.Instance))
+                // Does the candidate have anything remotely equivalent?
+                if (!candidate.Item3.Any(c => !c.IsErrorType() && bestBinding.Item3.Contains(c, LinkedFilesSymbolEquivalenceComparer.Instance)))
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs]: Test for the IsErrorType case?

Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs]: IMO using .Intersect(...).Any() would be clearer.

File: src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs [hunk 2]
```diff
@@ -104,10 +104,10 @@ internal abstract partial class AbstractSemanticQuickInfoProvider : AbstractQuic
 
             // We calculate the set of supported projects
             candidateResults.Remove(bestBinding);
-
             foreach (var candidate in candidateResults)
             {
-                if (!candidate.Item3.SequenceEqual(bestBinding.Item3, LinkedFilesSymbolEquivalenceComparer.Instance))
+                // Does the candidate have anything remotely equivalent?
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Providers/AbstractSemanticQuickInfoProvider.cs]: Nit: On first reading of this comment I thought it meant that you actually loosened the equivalence check on individual items ("anything *remotely* equivalent"). But I think you meant for the emphasis to be on the *anything* instead, right? If so, consider getting rid of the word "remotely" there.

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**